### PR TITLE
Implement job runner with priority queue

### PR DESCRIPTION
**Pull request by dctenna**

### CODE REVIEW

File: 08_algorithms_and_data_structures_1/PriorityQueue.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  enqueue(job: Job) {
+    const currentPriorityJobs = this.jobsByPriority[job.priority];
+    if (currentPriorityJobs) {
+      this.jobsByPriority[job.priority] = [job, ...currentPriorityJobs];
+    } else {
+      this.jobsByPriority[job.priority] = [job];
+    }
+  }
+
+  dequeue(): Job {
```
Comment (AlexHladin) [inline 08_algorithms_and_data_structures_1/PriorityQueue.ts:19]: @dctenna, the `.pop` method returns element or undefined. I think the proper return type of the `dequeue` function is `Job | undefined`

File: 08_algorithms_and_data_structures_1/main.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  queue.enqueue(newJob);
+}
+
+function getPriority(min: number, max: number) {
+  return Math.floor(Math.random() * (max - min + 1) + min);
+}
+
+const jobRunner = new JobRunner(queue);
+jobRunner.runJobs();
```
Comment (AlexHladin) [inline 08_algorithms_and_data_structures_1/main.ts:24]: code fails when I tried to run it :(

image

**Review by AlexHladin — CHANGES_REQUESTED** [review]

File: 08_algorithms_and_data_structures_1/PriorityQueue.ts (hunk 1 above)
Comment (dctenna) [reply 08_algorithms_and_data_structures_1/PriorityQueue.ts]: @AlexHladin done

File: 08_algorithms_and_data_structures_1/main.ts (hunk 2 above)
Comment (dctenna) [reply 08_algorithms_and_data_structures_1/main.ts:24]: @AlexHladin fixed to run with ts-node as well.

**Review by AlexHladin — APPROVED** [review]

**Pull request merged**